src/engine/Service.cpp

```diff
@@ -143,30 +143,30 @@ ProtoResult Service::computeResultImpl([[maybe_unused]] bool requestLaziness) {
       serviceQuery, "application/sparql-query",
       "application/sparql-results+json");
 
-  auto throwErrorWithContext = [&response, &serviceUrl](std::string_view sv) {
+  auto throwErrorWithContext = [this, &response](std::string_view sv) {
     std::string ctx;
     ctx.reserve(100);
-    for (const auto& s : std::move(response.body_)) {
-      ctx += reinterpret_cast<const char*>(s.data());
+    for (const auto& bytes : std::move(response.body_)) {
+      ctx += std::string(reinterpret_cast<const char*>(bytes.data()),
+                         bytes.size());
       if (ctx.size() >= 100) {
         break;
       }
     }
-
-    throw std::runtime_error(absl::StrCat(
-        "Error while executing a SERVICE request to <", serviceUrl.asString(),
-        ">: ", sv, ". First 100 bytes of the response: ",
-        ctx.substr(0, std::min(100, (int)ctx.size()))));
+    this->throwErrorWithContext(sv,
+                                ctx.substr(0, std::min(100, (int)ctx.size())));
```

Comment:
```suggestion
                                ctx.substr(0, 100));
```
`substr` automatically handles the out-of-bounds case.